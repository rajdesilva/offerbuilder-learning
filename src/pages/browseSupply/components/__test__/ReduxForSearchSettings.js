export const ReduxForSearchSettings = {
  searchparams: {
    destination: {},
    los: 1,
    distance: 10,
    onlySupplier: false,
    target: {
      suppliers: [],
      channels: [],
    },
    brands: [
      {
        storefronts: [{}],
      },
    ],
    dateRange: {
      startDate: "05/17/2021",
      endDate: "07/17/2021",
    },

    lcn: false,
    remainingCapitalPool: "",
    sortBy: {
      sortOrder: "ASCENDING",
      sortCriteria: "NAME",
    },
  },
  channelinfo: {
    loading: false,
    channels: [
      {
        id: "PCT",
        name: "PCT",
      },
      {
        id: "PCT",
        name: "ATG",
      },
      {
        id: "DEMO_OFFERBUILDER",
        name: "DEMO_OFFERBUILDER",
      },
    ],
  },
  sourceinfo: {
    loading: false,
    brands: [
      {
        id: "idb",
        name: "Internal Demo Brand",
        storefronts: [
          {
            id: "ids",
            name: "Internal Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
      {
        id: "cdb",
        name: "Client Demo Brand",
        storefronts: [
          {
            id: "cds",
            name: "Client Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
};
