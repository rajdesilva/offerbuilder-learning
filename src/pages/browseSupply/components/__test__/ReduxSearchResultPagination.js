export const PaginationStore = {
  searchedproperties: {
    loading: false,
    properties: [{}],
    totalMatch: 15,
  },
  searchparams: {
    pageOffset: 0,
    pageSize: 10,
  },
};
