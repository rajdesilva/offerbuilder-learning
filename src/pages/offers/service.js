import { baseURL } from "../../helpers";
import {
  getParsedOffers,
  getSearchAndFilterImagesQueryUrl,
  getSearchAndFilterQueryURL,
  updateReduxWithOfferData,
} from "../../helpers/utility";
import { store } from "../../redux/store";
import {
  marketingActions,
  newOfferActions,
  offerListSearchAndFilterActions,
} from "./actions";
import { offerImageSearchAndFilterActions } from "./actions/offerImageSearchAndFilterActions";

export const validateOfferId = (offerId) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}/exist`)
    .then((response) => {
      return response;
    })
    .catch((error) => {
      return error.toString();
    });
};

export const getCloudinarySignature = () => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}cloudinary/signature`)
    .then((response) => {
      return response;
    })
    .catch((error) => {
      return error.toString();
    });
};

export const getCityFromLatLong = (latLong) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${latLong.lat},${latLong.lng}&sensor=true&key=AIzaSyDydZJ7RZF1SAXwg4pZqnFFskYL8CW5RqA`;
  return fetch(url)
    .then((response) => response.json())
    .then((response) => {
      return response;
    })
    .catch((error) => {
      return error;
    });
};

export const getLanguages = () => {
  store.dispatch({
    type: marketingActions.FETCH_LANGUAGE_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}languages`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: marketingActions.FETCH_LANGUAGE_SUCCESS,
          payload: window.getValue(response, "data.languages"),
        });
      } else {
        store.dispatch({
          type: marketingActions.FETCH_LANGUAGE_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: marketingActions.FETCH_LANGUAGE_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const createNewOffer = (newOfferInfo) => {
  const url = `${baseURL.API_URL_BASE}offers`;
  store.dispatch({
    type: newOfferActions.CREATE_NEW_OFFER_LOADING,
    payload: true,
  });
  return window
    .fetchWrapper(url, {
      method: "POST",
      body: JSON.stringify(newOfferInfo),
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};

export const saveEditedOffer = (updatedOfferInfo) => {
  const url = `${baseURL.API_URL_BASE}offers/${updatedOfferInfo.id}`;
  return window
    .fetchWrapper(url, {
      method: "PATCH",
      body: JSON.stringify(updatedOfferInfo),
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};

export const getOfferDetails = (offerId) => {
  store.dispatch({
    type: offerListSearchAndFilterActions.OFFER_DETAILS_LOADING,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}`)
    .then((response) => {
      if (response.success) {
        let offerData = window.getValue(response, "data.offer");
        offerData = getParsedOffers(offerData);
        updateReduxWithOfferData(offerData);
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_DETAILS_SUCCESS,
        });
      } else {
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_DETAILS_FAILURE,
        });
      }
    })
    .catch((error) => {
      console.log(error.toString());
      store.dispatch({
        type: offerListSearchAndFilterActions.OFFER_DETAILS_FAILURE,
      });
    });
};

export const deleteOffer = (offerId) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}`, {
      method: "DELETE",
    })
    .then((response) => response);
};

export const updateOffer = (offerId, status) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}/${status}`, {
      method: "PUT",
    })
    .then((response) => response);
};

export const searchAndFilterOffer = (offerType) => {
  store.dispatch({
    type: offerListSearchAndFilterActions.OFFER_LIST_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}offers/search/` +
        getSearchAndFilterQueryURL(offerType)
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_LIST_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_LIST_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      console.error("error =", error.toString());
      store.dispatch({
        type: offerListSearchAndFilterActions.OFFER_LIST_FAILURE,
        payload: [],
      });
    });
};

export const getImagesFromProperties = (propertyCode) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}search?propertyCode=${propertyCode}`)
    .then((response) => response);
};

export const searchAndFilterImages = () => {
  store.dispatch({
    type: offerImageSearchAndFilterActions.OFFER_IMAGE_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}images/search` +
        getSearchAndFilterImagesQueryUrl()
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: offerImageSearchAndFilterActions.OFFER_IMAGE_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: offerImageSearchAndFilterActions.OFFER_IMAGE_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      console.error("error =", error.toString());
      store.dispatch({
        type: offerImageSearchAndFilterActions.OFFER_IMAGE_FAILURE,
        payload: [],
      });
    });
};

export const uploadImage = (file) => {
  let myHeaders = new Headers();

  myHeaders.append("Accept", "application/json");
  let formdata = new FormData();
  formdata.append("file", file);
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}images/upload`, {
      method: "POST",
      headers: myHeaders,
      body: formdata,
    })
    .then((response) => response);
};
