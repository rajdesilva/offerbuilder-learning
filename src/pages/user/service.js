import { baseURL } from "../../helpers";
import { userActions } from "./actions/userActions";
import { store } from "../../redux/store";

export const fetchUserInfo = () => {
  const url = `${baseURL.API_URL_BASE}user/`;
  store.dispatch({
    type: userActions.FETCH_USER_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(url)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: userActions.FETCH_USER_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: userActions.FETCH_USER_FAILURE,
          payload: false,
        });
      }
    })
    .catch(() => {
      store.dispatch({
        type: userActions.FETCH_USER_FAILURE,
        payload: false,
      });
    });
};

export const getUsersList = () => {
  store.dispatch({
    type: userActions.FETCH_USER_MANAGEMENT_LIST_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}users/`
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: userActions.FETCH_USER_MANAGEMENT_LIST_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: userActions.FETCH_USER_MANAGEMENT_LIST_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: userActions.FETCH_USER_MANAGEMENT_LIST_FAILURE,
        payload: [],
      });
    });
};

export const addNewUser = (newUserInfo) => {
  const url = `${baseURL.API_URL_BASE}users/user`;
  return window
    .fetchWrapper(url, {
      method: "POST",
      body: JSON.stringify(newUserInfo),
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};

export const deleteUser = (email, userRole) => {
  const url = `${baseURL.API_URL_BASE}users/user/${email}?roleNames=${userRole}`;
  return window
    .fetchWrapper(url, {
      method: "DELETE",
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};
