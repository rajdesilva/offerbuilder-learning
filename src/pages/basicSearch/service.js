import { browseSupplyActions, supplySearchActions } from "./actions";
import { store } from "./../../redux/store";
import { globalActions } from "../../redux/globalActions";
import { baseURL } from "../../helpers";
import {
  createSearchQueryURL,
  getSearchParamsToFetchMarketPrice,
} from "../../helpers/utility";
import { offerListSearchAndFilterActions } from "../offers/actions";

export const searchSupply = (routePostSearch, isEditFlow) => {
  store.dispatch({
    type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_LOADING,
    payload: true,
  });
  store.dispatch({
    type: supplySearchActions.SET_INCLUDE_MARKET_PRICE_FLAG,
    payload: false,
  });

  window
    .fetchWrapper(`${baseURL.API_URL_BASE}search/` + createSearchQueryURL(routePostSearch))
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_SUCCESS,
          payload: window.getValue(response, "data.properties"),
          totalMatch: window.getValue(response, "data.totalMatchedProperties"),
          pageOffset: window.getValue(response, "data.pageOffset"),
          pageSize: window.getValue(response, "data.pageSize"),
        });
      } else {
        store.dispatch({
          type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_FAILURE,
          payload: window.getValue(response, "error.apierror.message"),
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_FAILURE,
        payload: error.toString(),
      });
      store.dispatch({
        type: globalActions.SET_ERROR,
        payload: error.toString(),
      });
    });
};

export const advancedSearch = (
  params,
  searchType,
  page,
  pageSize,
  sortBy,
  order,
  groupType
  ) => {
    return window.fetchWrapper(`${baseURL.API_URL_BASE}global/properties?address=${params.address}&city=${params.city}&countryCode=${params.countryCode}&distance=${params.distance}&hotelName=${params.hotelName}&latitude=${params.latitude}&longitude=${params.longitude}&zip=${params.zip}&pageOffset=${page}&pageSize=${pageSize}&propertyGroupType=${groupType}&sortCriteria=${sortBy}&sortOrder=${order}&globalSearchType=${searchType}`);

    //return window.fetchWrapper(`https://offerbuilder-api-dev.kognitiv.com/global/properties?distance=10&globalSearchType=BASIC&latitude=47.2816&longitude=15.48809&propertyGroupType=ALL`);
}

export const globalSearchResult =(params)=>{
  
  window
    .fetchWrapper(`/global/properties?address=${params.address}`)
    .then((response) => {
        console.log(response)
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
        payload: [],
      });
      return [];
    });
}

export const getChannels = () => {
  store.dispatch({
    type: supplySearchActions.SEARCH_CHANNELS_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}portals`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.SEARCH_CHANNELS_SUCCESS,
          payload: window.getValue(response, "data.result"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getBrands = async () => {
  store.dispatch({
    type: supplySearchActions.SEARCH_BRANDS_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}sources`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.SEARCH_BRANDS_SUCCESS,
          payload: window.getValue(response, "data.brands"),
        });
        store.dispatch({
          type: offerListSearchAndFilterActions.UPDATE_OFFER_SELECTORS,
          payload: window.getValue(response, "data.brands"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.SEARCH_BRANDS_FAILURE,
          payload: [],
        });
      }
      return window.getValue(response, "data.brands");
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_BRANDS_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getCurrencies = async () => {
  store.dispatch({
    type: supplySearchActions.FETCH_CURRENCIES_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}currencies`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.FETCH_CURRENCIES_SUCCESS,
          payload: window.getValue(response, "data.currencies"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.FETCH_CURRENCIES_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.FETCH_CURRENCIES_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getMarketPricesForListOfProperties = () => {
  store.dispatch({
    type: supplySearchActions.FETCH_MARKET_PRICE_LOADING,
    payload: true,
  });

  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}search/marketprice/` +
        getSearchParamsToFetchMarketPrice()
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.FETCH_MARKET_PRICE_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.FETCH_MARKET_PRICE_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.FETCH_MARKET_PRICE_FAILURE,
        payload: [],
      });
      return [];
    });
};
