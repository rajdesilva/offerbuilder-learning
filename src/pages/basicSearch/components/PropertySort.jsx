import { Select } from "antd";
import React, { Fragment } from "react";
import { FormattedMessage } from "react-intl";
import { useDispatch } from "react-redux";
import { useHistory } from "react-router";
import { appConstants_results } from "../../../common";
import { supplySearchActions } from "../actions";
import { searchSupply } from "../service";
import styles from "./css/PropertySort.module.less";

const { Option } = Select;

export function PropertySort(props) {
  const dispatch = useDispatch();
  const history = useHistory();

  return (
    <Fragment>
      
      <Select span={3}
        className={styles.sltBox}
        style={{fontSize: '14px'}}
        data-testid="property-sort-order-by"
        dropdownMatchSelectWidth={false}
        defaultValue="ALL"
        onChange={(value) => {
          console.log(value)
          dispatch({
            type: supplySearchActions.SUPPLY_SEARCH_UPDATE_SORT_BY_GBL,
            payload: value,
          });
        }}
      >
       <option value="ALL"><FormattedMessage id="nemo.result.allproperties" /></option>
       <option value="NTP_SUPPLIER"><FormattedMessage id="nemo.result.withoutsuppliers" /></option>
       <option value="WITH_SUPPLIER"><FormattedMessage id="nemo.withsuppliers" /></option>
      </Select>{" "}
    </Fragment>
  );
}
