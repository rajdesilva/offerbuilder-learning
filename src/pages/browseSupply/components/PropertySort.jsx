import { Select } from "antd";
import React, { Fragment } from "react";
import { FormattedMessage } from "react-intl";
import { useDispatch } from "react-redux";
import { useHistory } from "react-router";
import { appConstants } from "../../../common";
import { supplySearchActions } from "../actions";
import { searchSupply } from "../service";
import styles from "./css/PropertySort.module.less";

const { Option } = Select;

export function PropertySort(props) {
  const dispatch = useDispatch();
  const history = useHistory();

  return (
    <React.Fragment>
      <span className={styles.sortBy}>
        <FormattedMessage id="nemo.sortBy" />
      </span>{" "}
      &nbsp;
      <Select
        data-testid="property-sort-order-by"
        dropdownMatchSelectWidth={false}
        defaultValue={JSON.stringify(props.sortBy)}
        onChange={(value) => {
          dispatch({
            type: supplySearchActions.SUPPLY_SEARCH_UPDATE_SORT_BY,
            payload: value,
          });
          searchSupply(history.location.pathname);
        }}
      >
        <Option
          data-testid="sort-name-ascending"
          value={JSON.stringify(
            appConstants.SORT_PROPERTY.NAME_ASCENDING_VALUE
          )}
        >
          {appConstants.SORT_PROPERTY.NAME_ASCENDING}
        </Option>
        <Option
          data-testid="sort-name-descending"
          value={JSON.stringify(
            appConstants.SORT_PROPERTY.NAME_DESCENDING_VALUE
          )}
        >
          {appConstants.SORT_PROPERTY.NAME_DESCENDING}
        </Option>
        <Option
          data-testid="sort-price-ascending"
          value={JSON.stringify(
            appConstants.SORT_PROPERTY.PRICE_ASCENDING_VALUE
          )}
        >
          {appConstants.SORT_PROPERTY.PRICE_ASCENDING}
        </Option>
        <Option
          data-testid="sort-price-descending"
          value={JSON.stringify(
            appConstants.SORT_PROPERTY.PRICE_DESCENDING_VALUE
          )}
        >
          {appConstants.SORT_PROPERTY.PRICE_DESCENDING}
        </Option>
      </Select>{" "}
      </React.Fragment>
  );
}
