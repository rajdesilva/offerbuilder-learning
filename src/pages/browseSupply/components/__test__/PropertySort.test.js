import React from "react";
import { createStore } from "redux";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import * as service from "./../../service";
import { render, fireEvent, act } from "../../../../helpers/testUtils";
import { PropertySort } from "../PropertySort";
import { tempStore } from "../../../offers/components/__test__/tempStore";

jest.mock("./../../service");
const history = createMemoryHistory();

describe("PropertySort component test cases", () => {
  test("PropertySort component snapshot", () => {
    const store = createStore(() => ({ ...tempStore }));
    const props = {
      sortBy: {
        sortOrder: "DESCENDING",
        sortCriteria: "PRICE",
      },
    };
    const wrapper = render(
      <Router history={history}>
        <PropertySort {...props} />
      </Router>,
      {
        store,
      }
    );
    const { container } = wrapper;
    expect(container.firstChild).toMatchSnapshot();
  });

  test("PropertySort - dispatch event when the option in sort is changed", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const props = {
      sortBy: {
        sortOrder: "DESCENDING",
        sortCriteria: "PRICE",
      },
    };
    const { getByTestId } = render(
      <Router history={history}>
        <PropertySort {...props} />
      </Router>,
      {
        store,
      }
    );

    service.searchSupply.mockImplementation(() => {
      return Promise.resolve({
        status: true,
      });
    });
    const searchSupplyFn = jest.spyOn(service, "searchSupply");
    const onPageChange = getByTestId("property-sort-order-by").querySelector(
      "input"
    );
    await act(async () => {
      await fireEvent.mouseDown(onPageChange);
      await fireEvent.click(getByTestId("sort-name-descending"));
      expect(searchSupplyFn).toHaveBeenCalled();
    });
  });
});
