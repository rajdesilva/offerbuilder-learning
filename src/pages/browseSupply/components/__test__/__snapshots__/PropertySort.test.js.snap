// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertySort component test cases PropertySort component snapshot 1`] = `
<span>
  Sort By
</span>
`;
